provider/nbd_image_export.py

```diff
@@ -0,0 +1,170 @@
+import logging
+
+from avocado.utils import process
+from avocado.core import exceptions
+
+from virttest import data_dir
+from virttest import qemu_storage
+from virttest import utils_qemu
+from virttest import utils_misc
+from virttest import qemu_devices
+
+
+class NBDExportImage(object):
+    """NBD local image export base class"""
+
+    def __init__(self, local_image, params):
+        """
+        Initialize object.
+        :param local_image: local image tag
+        :param params: dictionary containing all test parameters.
+        """
+        self._tag = local_image
+        self._params = params
+        self._image_params = self._params.object_params(self._tag)
+
+    def create_image(self):
+        result = None
+        if self._image_params.get('create_image_cmd'):
+            result = process.run(self._image_params['create_image_cmd'],
+                                 ignore_status=True, shell=True)
+        elif not self._image_params.get_boolean("force_create_image"):
+            _, result = qemu_storage.QemuImg(self._image_params,
+                                             data_dir.get_data_dir(),
+                                             self._tag).create(self._params)
+
+        if result and result.exit_status != 0:
+            raise exceptions.TestFail('Failed to create image, error: %s'
+                                      % result.stderr.decode())
```

Comment:
Do we need to assign ` result = None`? 
if need it, please add checking `if result is None:`, since if the `result is None`, actually, `create_image` will do nothing for creating, I don't know whether you wanna it does nothing for this scenario.
if not, we should delete it.

Reply:
it's a kind of 'var namespace'  coming from c/c++, we define a var inside a block (e.g. if block) we cannot ref it outside the block, looks python will search it inside the blocks, so remove the line.